drivers: usb: device: stm32: Configure USB Prescaler for F103

kuroneko opened the pull request:
- If the HAL headers expose the USBPRE flag, then we're probably
  dealing with a F103 - using the same fundamental logic as the
  code for the OTG models, set-up the USB Prescaler correctly.

- Fixes #47146

Commit:
drivers: usb: device: stm32: Configure USB Prescaler for F103

- If the HAL headers expose the USBPRE flag, then we're probably
  dealing with a F103 - using the same fundamental logic as the
  code for the OTG models, set-up the USB Prescaler correctly.

- Fixes #47146

Signed-off-by: Chris Collins <chris@realsimgear.com>

erwango reviewed (dismissed):
I theory I would push to use clock_control  API to switch clock source, but F1 is one of the series where this is not yet available, so no reason to bother.
Approved

FRASTM commented on drivers/usb/device/usb_dc_stm32.c:
```diff
@@ -293,8 +293,28 @@ static int usb_dc_stm32_clock_enable(void)
+		case 48000000U:
+			LL_RCC_SetUSBClockSource(LL_RCC_USB_CLKSOURCE_PLL);
+			break;
+		case 72000000U:
```
case MHZ(72):

kuroneko replied:
Fixed this, as well as the original instance in the source in the OTG initialisation.

FRASTM commented on drivers/usb/device/usb_dc_stm32.c:
```diff
@@ -293,8 +293,28 @@ static int usb_dc_stm32_clock_enable(void)
+	 */
+	if (LL_RCC_GetSysClkSource() == LL_RCC_SYS_CLKSOURCE_STATUS_PLL) {
+		switch (sys_clock_hw_cycles_per_sec()) {
+		case 48000000U:
```
could be written as `case MHZ(48):`

kuroneko replied:
Fixed this, as well as the original instance in the source in the OTG initialisation.

Commit:
drivers: usb: device: stm32: Use macros for frequencies

Rather than explicitly setting out the frequnecy in full, use the MHZ()
macro.

Signed-off-by: Chris Collins <chris@realsimgear.com>
